tests/integration/test_mixed_space_bayesian_optimization.py

```diff
@@ -234,11 +234,50 @@ def objective(x: TensorType) -> TensorType:
     )
 
 
+def _get_categorical_problem() -> SingleObjectiveTestProblem[TaggedProductSearchSpace]:
+    # a categorical scaled branin problem with 6 categories mapping to 3 random points
+    # plus the 3 minimizer points (to guarantee that the minimum is present)
+    points = tf.concat(
+        [tf.random.uniform([3], dtype=tf.float64), ScaledBranin.minimizers[..., 0]], 0
+    )
+    return categorical_scaled_branin(tf.random.shuffle(points))
+
+
+cat_problem = _get_categorical_problem()
+
+
 @random_seed
 @pytest.mark.parametrize(
     "num_steps, acquisition_rule",
     [
         pytest.param(25, EfficientGlobalOptimization(), id="EfficientGlobalOptimization"),
+        pytest.param(
```

Comment:
Is it worth adding a `FixedPointTrustRegionDiscrete` test as well, like the discrete-numerical case?

Also, is it possible to combine `test_optimizer_finds_minima_of_the_categorical_scaled_branin_function` with `test_optimizer_finds_minima_of_the_scaled_branin_function`, so we don't have to maintain them separately?

Reply:
I've merged them now using a helper function. The tests are reasonably slow so my preference is not to add a categorical `FixedPointTrustRegionDiscrete` test unless it's testing an important combination.